### PR TITLE
feat(sdk): extract abstract connection and add username-password connection

### PR DESCRIPTION
**Pull request by jackworkgit**
1. Extract abstract connection using template pattern.
2. Add username-password connection

### CODE REVIEW

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/dto/UpdateConnectionDto.java [hunk 1]
```diff
@@ -2,16 +2,17 @@
...
+import jakarta.validation.constraints.NotNull;
 import lombok.Data;
 
 @Data
 public class UpdateConnectionDto {
 
   @NotBlank private String id;
 
-  private ConnectionStrategryEnum strategy = ConnectionStrategryEnum.SAML;
+  @NotNull private ConnectionStrategryEnum strategy;
```
Comment (DaveXiong) [inline vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/dto/UpdateConnectionDto.java:13]: Can change `strategy` on updating request ?
if not, may need to remove `strategy`

**Review by DaveXiong — COMMENTED** [review]
and pls help to fix the code smells reported by SonarCloud

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/dto/UpdateConnectionDto.java (hunk 1 above)
Comment (jackworkgit) [reply vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/dto/UpdateConnectionDto.java:13]: ok

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/dto/UpdateConnectionDto.java [hunk 2]
```diff
@@ -2,16 +2,17 @@
...
 
-  private OidcConnection odic;
+  private OidcConnectionDto odic;
```
Comment (pori) [inline vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/dto/UpdateConnectionDto.java:15]:
```suggestion
  private OidcConnectionDto oidc;
```

Probably warrants more replacements but I can see this getting very confusing elsewhere.

**Review by pori — DISMISSED** [review]

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/dto/UpdateConnectionDto.java (hunk 2 above)
Comment (jackworkgit) [reply vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/dto/UpdateConnectionDto.java:15]: This parameter represents the open-id connection that receives parameters passed in by the front end and will be used in specific connections.

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/dto/UpdateConnectionDto.java (hunk 1 above)
Comment (DaveXiong) [reply vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/dto/UpdateConnectionDto.java:13]: As  the connection's id has been supplied, you should use the `id` to query the connection details, based on the connection details, you should be know the conneciton's strategy.

Comment (jackworkgit) [reply vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/dto/UpdateConnectionDto.java:13]: Ok, I will delete it.

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/dto/UpdateConnectionDto.java (hunk 2 above)
Comment (pori) [reply vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/dto/UpdateConnectionDto.java:15]: The variable is still mispelled.

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/dto/UpdateConnectionDto.java (hunk 1 above)
Comment (jackworkgit) [reply vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/dto/UpdateConnectionDto.java]: done

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/dto/UpdateConnectionDto.java (hunk 2 above)
Comment (jackworkgit) [reply vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/dto/UpdateConnectionDto.java]: Changed it to `openID`.

**Review by pori — APPROVED** [review]

**Pull request merged**